src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -123,6 +123,14 @@ count_dots (const char *s)
    return n;
 }
 
+/* return a lowercased string.  must be freed */
+static char *
+lowercase_const_str(const char *key)
```

Comment:
Throughout our codebase we the suffix `_new` in function names to signal to other developers that whatever the function returns is _newly allocated_ and must be freed. That being said, if this was renamed to something like `lowercase_str_new` we wouldn't need the comment above. Since the functions' name would signal: 1) it's returning a lowercase str 2) it needs to be freed. But I'm fine with the way it is currently too.